### PR TITLE
Adds rudimentary Dutch parsing support

### PR DESCRIPTION
**Pull request by mikewoudenberg**
## Description
Adds rudimentary Dutch language support to Mycroft parser as mentioned in #2380 

## How to test
- Run the unit tests
- Change Mycroft's language to Duth and experiment with any skill using the internal parser. These skills use the parser:
    - The cut up skill
    - The dice-skill
    - The wolfram alpha skill
    - The mycroft timer skill
    - The mycroft wink iot skill
    - The alarm skill
    - The audio record skill
    - The date-time skill
    - The reminder skill
    - The volume skill
    - The weather skill

## Contributor license agreement signed?
CLA [x]

### CODE REVIEW

**Comment by devs-mycroft** [comment]
Hello, @mikewoudenberg, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

File: mycroft/util/lang/common_data_nl.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Mycroft AI Inc.
```
[the remark this answers is not in the document]
Comment (mikewoudenberg) [reply mycroft/util/lang/common_data_nl.py]: Done

File: mycroft/util/lang/parse_nl.py [hunk 2]
```diff
@@ -0,0 +1,1480 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Mycroft AI Inc.
```
[the remark this answers is not in the document]
Comment (mikewoudenberg) [reply mycroft/util/lang/parse_nl.py]: Done

File: mycroft/util/lang/common_data_nl.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    (1e120, "vigintillion"),
+    (1e306, "unquinquagintillion"),
+    (1e312, "duoquinquagintillion"),
+    (1e336, "sesquinquagintillion"),
+    (1e366, "unsexagintillion")
+])
+
+
+_SHORT_SCALE_NL = OrderedDict([
+    (100, 'hundred'),
```
Comment (j1nx) [inline mycroft/util/lang/common_data_nl.py]: 100 till 1e15 are not yet translated. Everything ening at *joen I consider fine, hardly used anyhow and about 99,99% of the Dutch people don't know them any way..

File: mycroft/util/lang/parse_nl.py [hunk 4]
```diff
@@ -0,0 +1,1480 @@
...
+
+# decimal marker ( 1 point 5 = 1 + 0.5)
+_DECIMAL_MARKER = {"komma", "punt"}
+
+_STRING_NUM_NL = _invert_dict(_NUM_STRING_NL)
+_STRING_NUM_NL.update({
+    "half": 0.5,
+    "driekwart": 0.75,
+    "anderhalf": 1.5,
+    "couple": 2
```
Comment (j1nx) [inline mycroft/util/lang/parse_nl.py]: Couple is still english. I think this best translate to "dubbel"

File: mycroft/util/lang/parse_nl.py [hunk 5]
```diff
@@ -0,0 +1,1480 @@
...
+    timeQualifierOffsets = [8, 15, 19, 0]
+    markers = ['op', 'in', 'om', 'tegen', 'over',
+               'deze', 'rond', 'voor', 'van', "binnen"]
+    days = ["maandag", "dinsdag", "woensdag", "donderdag", "vrijdag",
+            "zaterdag", "zondag"]
+    day_parts = [a + b for a in days for b in timeQualifiersList]
+    months = ['januari', 'februari', 'maart', 'april', 'mei', 'juni',
+              'juli', 'augustus', 'september', 'oktober', 'november',
+              'december']
+    recur_markers = days + [d+'en' for d in days] + ['weekend', 'weekday',
```
Comment (j1nx) [inline mycroft/util/lang/parse_nl.py]: weekend - weekeinde
weekday - werkdag

File: mycroft/util/lang/parse_nl.py [hunk 6]
```diff
@@ -0,0 +1,1480 @@
...
+                                                     'weekends', 'weekdays']
```
Comment (j1nx) [inline mycroft/util/lang/parse_nl.py]: weekends - Weekeinden
weekdays - werkdagen

File: mycroft/util/lang/parse_nl.py [hunk 7]
```diff
@@ -0,0 +1,1480 @@
...
+            elif wordNextNext == "minuten":
+                minOffset = 2
+            elif wordNextNext == "seconden":
+                secOffset = 2
+        # parse half an hour, quarter hour
+        elif word == "uur" and \
+                (wordPrev in markers or wordPrevPrev in markers):
+            if wordPrev == "half":
+                minOffset = 30
+            elif wordPrev == "quarter":
```
Comment (j1nx) [inline mycroft/util/lang/parse_nl.py]: quarter - kwartier

File: mycroft/util/lang/parse_nl.py [hunk 8]
```diff
@@ -0,0 +1,1480 @@
...
+                minOffset = 15
+            elif wordPrevPrev == "quarter":
```
Comment (j1nx) [inline mycroft/util/lang/parse_nl.py]: see above

File: mycroft/util/lang/parse_nl.py (hunk 4 above)
Comment (mikewoudenberg) [reply mycroft/util/lang/parse_nl.py]: I opted for "Paar" as the english parser was tailored to recognize "Couple of hours" which translates to the Dutch "Paar uur"

**Comment by krisgesling** [comment]
Thanks Mike, CLA received. So this is good to go if all requests have been addressed.

**Comment by forslund** [comment]
When @j1nx thinks this is good to go I'll hit the Merge button!

**Comment by j1nx** [comment]
Looks fine by me. Meaning looks fine translation-wise from a second Dutch guy's opinion / review.

Not 100% sure if all the big numbers translates correctly, but to be honest, didn't even know they  exists ;)

Glad somebody picked this up, so am all for merging. This means we can use Google STT/TTS and change to Dutch.

Thank you very much for this work @mikewoudenberg

**Comment by forslund** [comment]
Many Thanks for this, Merging now!

**Pull request merged**